src/Util/Help.php

```diff
@@ -0,0 +1,626 @@
+     * @return void
+     */
+    public function display()
+    {
+        $this->printUsage();
+        $this->printCategories();
+
+    }//end display()
+
+
+    /**
+     * Filter the available options based on the requested options.
+     *
+     * @return void
+     */
+    private function filterOptions()
+    {
+        $filteredOptions = $this->getAllOptions();
+
+        foreach ($filteredOptions as $category => $options) {
+            // Initial state set to "true" to prevent a spacer at the start of an array.
+            $lastWasSpacer = true;
+            $spacerCount   = 0;
+
+            foreach ($options as $name => $option) {
+                if ($lastWasSpacer !== true && strpos($name, 'blank-line') === 0) {
+                    ++$spacerCount;
+                    $lastWasSpacer = true;
+                    continue;
+                }
+
+                if (in_array($name, $this->requestedOptions, true) === false) {
+                    unset($filteredOptions[$category][$name]);
+                    continue;
+                }
+
+                $lastWasSpacer = false;
+            }
+
+            // Make sure the final array doesn't contain a spacer at the end.
+            if (empty($filteredOptions[$category]) === false) {
+                end($filteredOptions[$category]);
+                $key = key($filteredOptions[$category]);
+                if (strpos($key, 'blank-line') === 0) {
+                    unset($filteredOptions[$category][$key]);
+                    --$spacerCount;
+                }
+            }
+
+            // Remove categories now left empty.
+            if (empty($filteredOptions[$category]) === true || count($filteredOptions[$category]) === $spacerCount) {
+                unset($filteredOptions[$category]);
+            }
+        }//end foreach
+
+        $this->activeOptions = $filteredOptions;
+
+    }//end filterOptions()
+
+
+    /**
+     * Determine the length of the longest option argument and store it.
+     *
+     * @return void
+     */
+    private function setMaxOptionNameLength()
+    {
+        $lengths = [];
+        foreach ($this->activeOptions as $category => $options) {
+            foreach ($options as $option) {
+                if (isset($option['argument']) === false) {
+                    continue;
+                }
+
+                $lengths[] = strlen($option['argument']);
+            }
+        }
+
+        if (empty($lengths) === false) {
+            $this->maxOptionNameLength = max($lengths);
+        }
+
+    }//end setMaxOptionNameLength()
+
+
+    /**
+     * Get the maximum width which can be used to display the help info.
+     *
+     * Independently of user preference/auto-determined width of the current screen,
+     * a minimum width is needed to display information, so don't allow this to get too low.
+     *
+     * @return int
+     */
+    private function getMaxWidth()
+    {
+        return max(self::MIN_WIDTH, $this->config->reportWidth);
```

Comment:
Can `$this->config->reportWidth` change during execution?

If not, then instead of this method, would it be worth creating another property, `$maxWidth` and setting this in `::__construct()` with `max(self::MIN_WIDTH, $this->config->reportWidth)`?

Reply:
> Can `$this->config->reportWidth` change during execution?

Not once this class gets instantiated (and PHPCS exits straight after). 

> If not, then instead of this method, would it be worth creating another property, `$maxWidth` and setting this in `::__construct()` with `max(self::MIN_WIDTH, $this->config->reportWidth)`?

I had that in an earlier draft of the class, but ended up with the method as I found it more descriptive for the other width calculations to have all these calculations in methods close together (so the logic is more comprehensible).